src/rp2_common/hardware_uart/include/hardware/uart.h

```diff
@@ -93,11 +82,13 @@ typedef struct uart_inst uart_inst_t;
 
 /** @} */
 
-#ifndef PICO_DEFAULT_UART_INSTANCE
+#if !defined(PICO_DEFAULT_UART_INSTANCE) && defined(PICO_DEFAULT_UART)
 #define PICO_DEFAULT_UART_INSTANCE (__CONCAT(uart,PICO_DEFAULT_UART))
 #endif
 
+#ifdef PICO_DEFAULT_UART_INSTANCE
 #define uart_default PICO_DEFAULT_UART_INSTANCE
```

Comment:
This may be a silly question, but why the separate `uart_default` define rather than just continuing to use `PICO_DEFAULT_UART_INSTANCE` ?

Reply:
just seemed friendlier